Sources/NIOAsyncAwaitDemo/AsyncChannelIO.swift

```diff
@@ -24,17 +24,21 @@ struct AsyncChannelIO<Request, Response> {
     }
 
     func start() async throws -> AsyncChannelIO<Request, Response> {
+        self.channel.eventLoop.preconditionNotInEventLoop()
```

Comment:
Hm yeah all those are cases where we'd want to stay on the EL... this is useful to point at as specific examples, thanks Cory!

I could propose unholy horrible workarounds, but let's not... 😅